### PR TITLE
Add 'grade points' grading scheme

### PR DESCRIPTION
**Pull request by steveswinsburg**
This addresses #3432 and adds a new grading scheme called Grade Points. See the issue for the full details.

### CODE REVIEW

**Comment by steveswinsburg** [comment]
I have been unable to verify the MySQL conversion as my db is borked and I only have an Oracle instance. Can someone please run it? Roll back afterwards, it is just a bunch of inserts.

File: reference/docs/conversion/sakai_12_mysql_conversion.sql [hunk 1]
```diff
@@ -246,3 +245,123 @@ ALTER TABLE SAKAI_POSTEM_STUDENT MODIFY COLUMN username varchar(99), DROP INDEX
 
 -- #3431 Add final grade mode setting
 ALTER TABLE GB_GRADEBOOK_T ADD COLUMN FINAL_GRADE_MODE bit(1) NOT NULL DEFAULT b'0';
+
+-- BEGIN 3432 Grade Points Grading Scale
+-- add the new grading scale
+INSERT INTO gb_grading_scale_t (id, object_type_id, version, scale_uid, name, unavailable)
+VALUES (gb_grading_scale_s.nextval, 0, 0, 'GradePointsMapping', 'Grade Points', 0);
```
Comment (payten) [inline reference/docs/conversion/sakai_12_mysql_conversion.sql]: Hm I hit `Unknown column 'gb_grading_scale_s.nextval' in 'field list'` when I run the first INSERT.  Perhaps I'm missing something?

Comment (steveswinsburg) [inline reference/docs/conversion/sakai_12_mysql_conversion.sql]: ahh thats the Oracle sequence. I have to change to the auto increment version.

File: edu-services/gradebook-service/impl/src/java/org/sakaiproject/component/gradebook/GradebookServiceHibernateImpl.java [hunk 2]
```diff
@@ -3260,6 +3260,9 @@ private Double calculateCategoryScore(String studentUuid, Long categoryId, List<
 			List<Assignment> assignments = getAssignmentsCounted(gradebook.getId());
 			GradeMapping gradeMap = gradebook.getSelectedGradeMapping();
 			
+			System.out.println("gradeMap: " + gradeMap);
+			System.out.println("percents: " + gradeMap.getDefaultBottomPercents());
```
Comment (payten) [inline edu-services/gradebook-service/impl/src/java/org/sakaiproject/component/gradebook/GradebookServiceHibernateImpl.java]: Should these be logging to debug only?

Comment (steveswinsburg) [inline edu-services/gradebook-service/impl/src/java/org/sakaiproject/component/gradebook/GradebookServiceHibernateImpl.java]: indubidibly

**Comment by steveswinsburg** [comment]
Ah MySQL and being case sensitive... Also fixed that.

**Comment by ottenhoff** [comment]
Is this meant to be merged into 11.x?

**Pull request merged**